### ISSUE TITLE
[Header] - With more than two levels of items the first menu doesn't appear as selected

### ISSUE THREAD

**Issue by andrefrua** [lumada-design/hv-uikit-react#3525]
### Which version of NEXT UI Kit are you opening this bug for?

v5.x

### Latest version

- [X] I have tested the latest version

### No Hitachi confidential content

- [X] I'm not including any Hitachi confidential content

### Current behavior 😯

Currently when sending more than 2 levels of items to the Header component the menu on the top level isn't getting selected as you can see in the image below:
![image](https://github.com/lumada-design/hv-uikit-react/assets/26412841/b9f0d4cb-a135-4f62-a493-3c0f5e165b77)

The navigation data used to replicate the issue is the following:
```
[
  {
    id: "1",
    label: "Overview",
    data: [
      {
        id: "1-1",
        label: "Model Effectiveness 1",
        href: "/overview/model-effectiveness",
      },
      {
        id: "1-2",
        label: "Trend Analysis 1-2",
        href: "/overview/trend-analysis",
      },
    ],
  },
  {
    id: "2",
    label: "Events",
    href: "/events",
  },
  {
    id: "3",
    label: "Work Orders",
    data: [
      {
        id: "3-1",
        label: "Model Effectiveness 3-1",
        data: [
          {
            id: "3-1-1",
            label: "Another Sub Level 3-1-1",
            href: "/work-orders/model-effectiveness-3-1-1",
          },
          {
            id: "3-1-2",
            label: "Another Sub Level 3-1-2",
            href: "/work-orders/trend-analysis-3-1-2",
          },
        ],
      },
      {
        id: "3-2",
        label: "Trend Analysis 3-2",
        href: "/work-orders/trend-analysis",
      },
    ],
  },
  {
    id: "4",
    label: "Assets",
    href: "/assets",
  },
  {
    id: "5",
    label: "Analytics",
    data: [
      {
        id: "5-1",
        label: "Model Effectiveness 5-1",
        href: "/analytics/model-effectiveness",
      },
      {
        id: "5-2",
        label: "Trend Analysis 5-2",
        href: "/analytics/trend-analysis",
      },
    ],
  },
];
```

### Expected behavior 🤔

When a sub item is selected the top menu should also appear selected

**Comment by zettca** [comment]
Sorry for the delay @andrefrua. We'll look into this 👍

**Comment by plagoa** [comment]
@andrefrua please check the PR here:
https://github.com/lumada-design/hv-uikit-react/pull/3600

Let us know if this fixes the issue.

**Comment by MEsteves22** [comment]
Hello 👋

Thank you for opening this issue. This was fixed on [v5.24.4](https://github.com/lumada-design/hv-uikit-react/releases/tag/%40hitachivantara%2Fuikit-react-core%405.24.4).
